File: packages/sdk-csharp/src/Primust/RecordResult.cs
namespace Primust;

/// <summary>
/// Returned by Run.RecordAsync().
/// Write CommitmentHash to your operational logs for log linkage.
/// </summary>
public sealed class RecordResult
{
    /// <summary>Unique record identifier.</summary>
    public string RecordId { get; init; } = "";

    /// <summary>
    /// Log linkage anchor — write to your operational logs alongside the
    /// transaction ID. Connects your logs to the VPEC.
    /// </summary>
    public string CommitmentHash { get; init; } = "";

    /// <summary>Output commitment hash (if output was provided).</summary>
    public string? OutputCommitment { get; init; }

    /// <summary>Hash algorithm used: "sha256" or "poseidon2".</summary>
    public string CommitmentAlgorithm { get; init; } = "sha256";

    /// <summary>Proof level achieved.</summary>
    public string ProofLevel { get; init; } = "attestation";

    /// <summary>ISO 8601 timestamp of recording.</summary>
    public string RecordedAt { get; init; } = "";

    /// <summary>Rolling chain integrity hash.</summary>
    public string ChainHash { get; init; } = "";

    /// <summary>True if API was unreachable — record queued locally.</summary>
    public bool Queued { get; init; }
}

File: sdk-java/src/main/java/com/primust/sdk/RecordResult.java
package com.primust.sdk;

/**
 * Returned by Run.record(). Write commitmentHash to your operational logs
 * for log linkage — connects your logs to the VPEC.
 */
public final class RecordResult {

    private final String recordId;
    private final String commitmentHash;
    private final String outputCommitment;
    private final String commitmentAlgorithm;
    private final String proofLevel;
    private final String recordedAt;
    private final String chainHash;
    private final boolean queued;

    public RecordResult(
            String recordId,
            String commitmentHash,
            String outputCommitment,
            String commitmentAlgorithm,
            String proofLevel,
            String recordedAt,
            String chainHash,
            boolean queued) {
        this.recordId = recordId;
        this.commitmentHash = commitmentHash;
        this.outputCommitment = outputCommitment;
        this.commitmentAlgorithm = commitmentAlgorithm;
        this.proofLevel = proofLevel;
        this.recordedAt = recordedAt;
        this.chainHash = chainHash;
        this.queued = queued;
    }

    public String recordId() { return recordId; }
    /** Log linkage anchor — write to your operational logs alongside transaction ID. */
    public String commitmentHash() { return commitmentHash; }
    public String outputCommitment() { return outputCommitment; }
    public String commitmentAlgorithm() { return commitmentAlgorithm; }
    public String proofLevel() { return proofLevel; }
    public String recordedAt() { return recordedAt; }
    public String chainHash() { return chainHash; }
    /** True if API was unreachable — record queued locally, will flush on reconnect. */
    public boolean queued() { return queued; }
}
